components/sensor/lc709203f.rst

```diff
@@ -0,0 +1,147 @@
+battery monitor with ESPHome. This device is available as a 
+`standalone sensor <https://www.adafruit.com/product/4712>`__ and it is also one of the battery
+monitor chips used on their `ESP32 Feather <https://www.adafruit.com/product/5477>`__ dev boards.
+
+.. image:: images/LC709203f_product_image.jpg
+    :width: 30.0%
+
+.. image:: images/LC709203f_feather_product_image.jpg
+    :width: 30.0%
+
+.. note:: 
+
+    This device does not contain a temperature sensor. Only enable the temperature sensor option 
+    if you have attached a thermistor as shown in :ref:`temperature-sensor`.
+
+Prerequisites:
+-----------------------------------------
+
+This component relies on the :ref:`I²C <i2c>` componenent to be setup. See the reference for
+that component for details. A basic example of what that looks like is below.
+
+.. code-block:: yaml
+
+    i2c:
+      sda: SDA
+      scl: SCL
+      scan: true
+      id: bus_a
+
+Configuration Example:
+-----------------------------------------
+An example configuration YAML is shown below.
+
+.. code-block:: yaml
+
+    sensor:
+      - platform: lc709203f
+        size: 2000
+        voltage: 3.7
+        battery_voltage:
+          name: "Battery Voltage"
+        battery_level:
+          name: "Battery"
+        temperature:
+          name: "Pack Temperature"
+          thermistor_b_constant: 0xA5A5
+
+Configuration variables:
+-----------------------------------------
+
+- **size** (*Optional*): Size of the battery in mAH. 
+
+  - Valid values are integers between 200 and 3000.
+  - Defaults to 500 mAH. You should really set this correctly but if you really don't know, this default is probably okay.
+  - see :ref:`pack-size-and-voltage` for help if you don't know your pack size.
+  
+- **voltage** (*Optional*): nominal voltage of the battery pack in V.
+
+  - Valid values are ``3.7`` or ``3.8``
+  - Defaults to ``3.7``. This is the correct value for the Adafruit batteries.
+  - see :ref:`pack-size-and-voltage` for help if you don't know your pack voltage.
+  - see :ref:`pack-voltage` for more information on how this value is used.
+
+- **battery_voltage** (*Optional*): The setup information for the voltage sensor.
+
+  - Standard options from :ref:`Sensor <config-sensor>`.
+  - **name** (*Optional*): I suggest setting this one.
+  
+- **battery_level** (*Optional*): The setup information for the battery level sensor.
+
+  - Standard options from :ref:`Sensor <config-sensor>`.
+  - **name** (*Optional*): I suggest setting this one.
+
+- **temperature** (*Optional*): The setup information for the battery temperature sensor.
+  
+  - **thermistor_b_constant** (*Optional*): The B-constant of the thermistor. Defaults 
+    to `0x0BA6` but if you are using a thermistor, you should really change this.
+  - Standard options from :ref:`Sensor <config-sensor>`.
+  - **name** (*Optional*): I suggest setting this one.
```

Comment:
As discussed on the code PR, let's make this optional -- but also, let's not refer to the B-constant in hexadecimal (I alluded to this in my comment on the code PR). I know the chip's datasheet does this, but given the use case, it just plain doesn't make sense to use hexadecimal notation. You're not going to find the B-constant in any thermistor's datasheet written in hexadecimal. 😉

```suggestion
- **temperature** (*Optional*): Configuration for the battery temperature sensor.
  
  - **thermistor_b_constant** (*Optional*): The B-constant of the thermistor. Defaults 
    to `2982` but should be set based on the thermistor you are using.
  - All options from :ref:`Sensor <config-sensor>`.
```